### DISCUSSION TITLE
Guias de estilo para o uso do GIT

### DISCUSSION THREAD

**Discussion by domeniqque-pereira-deel** [frontendbr/forum#1180] [Perguntas e Respostas]
Olá pessoal! 

Já passei no início pela fase de fechar o olho e fazer o commit (git add --a && gc -m "altas alterações" rs), mas me dei mal algumas vezes tentando achar onde havia feito certa alteração. Ou então precisar voltar algum commit e não saber qual, correndo o risco de voltar coisa que não devia. Porém, assisti uma talk no trabalho sobre GIT e boas práticas e decidi parar de sofrer. 
 
Agora, acho muito válido (e bonito 😍) o uso de um guia de estilo, mesmo que a equipe não siga um padrão de estilo de mensagens de commit, nomeação de branch, etc. ou ainda em projetos pessoais.

Lembro que havia feito umas pesquisas sobre GIT style guides, porém pareceu um pouco difícil encontrá-los. Pelo que percebi, a maioria possui pontos em comum. Mas ainda acho que seria interessante uma referência (mesmo que breve) sobre o assunto. 

Gostaria de saber se você ou sua equipe utiliza algum guia de estilo para GIT? Ou sua opinião sobre o assunto. Thank's! ;)

**Comment by domeniqque-pereira-deel** [comment]
Gostei muito do guia de commit msg usado pela equipe do [Karma](http://karma-runner.github.io/2.0/dev/git-commit-msg.html)  e os tópicos sobre **branch**, **PR**, **fetching** e **merging**  feitos pelo pessoal da [Leroy Merlin](https://github.com/leroy-merlin-br/garden/blob/master/CONTRIBUTING.md). A um tempo venho usando essa combinação. 

Outros que achei interessante (e parecidos):

* [Udacity](http://udacity.github.io/git-styleguide)
* [Pagar.me](https://github.com/pagarme/git-style-guide)

**Comment by dougaraujos** [comment]
Faz uns anos criei essa thread sobre template de commit message: https://gibbon.co/dougaraujos/git-best-practices-commit-messages

**Comment by domeniqque-pereira-deel** [comment]
Legal @dougaraujos, interessante o item **2. Guidelines & Examples of good & bad practices**. Bem extenso

**Comment by didiraja** [comment]
Nos dois trabalhos, sempre bastou ser explicativo. sem uso de template específico.

**Comment by julioxavierr** [comment]
A gente tem um guideline bem legal aqui na Entria, também usamos `commitlint` pra assegurar que o padrão seja cumprido por todos, funciona bem.
É open source, dá uma olhada [aqui](https://github.com/entria/entria-guidelines/blob/master/git-guideline.md).
